cf_remote/aramid.py

```diff
@@ -152,11 +153,21 @@ def communicate(self, timeout=1, ignore_failed=False):
                             % (self.host.host_name, self._max_retries)
                         )
                     else:
-                        raise _TaskError("SSH failed on '%s'" % self.host.host_name)
+                        raise _TaskError(
+                            "SSH failed on '%s' with error '%s'"
+                            % (self.host.host_name, self.stderr)
+                        )
             else:
                 self.done = True
                 self.stdout += out
                 self.stderr += err
+                if self.proc.returncode != 0:
+                    print(
+                        "Command '%s' failed with return code %d stderr was '%s'"
+                        % (" ".join(self.proc.args), self.proc.returncode, self.stderr)
+                    )
```

Comment:
@craigcomstock I believe this is not what we want, i.e. we want to be able to run the function with a command that might return non-zero, and we want to be able to handle that nicely in the place where we call this, rather than this automatically printing potentially noisy errors.

Reply:
Exactly. `aramid` is supposed to be a module, not a one-off thing tailored to a very specific use case when `print()` inside of a function is acceptable.

Reply:
I didn't see the errors going upstream which is a bug in aramid in my opinion, however, I will try harder to make sure that the error and stderr make their way out and we catch bits as needed in the calling code so that aramid doesn't print out.

I agree that it should not print but didn't see another way, yet...